### PR TITLE
Clean export code

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
Working towards duplication removal

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
Status: 

* `Special:VerifiedExport` was removed. `Special:Export` now always includes the added info. Or at least it should. I have not extensively tested this yet and the single PHPUnit test is insufficient yet to instill much confidence. If we want to make the extra info optional, we still need to implement the UI for that somehow.

* The default of `Include only the current revision, not the full history` need to be changed back to `not checked`

* PHP 8 is now used. Changes made against MW 37, so older versions might no longer work.

* Style checks and static analysis added to the CI. Can also be run with `make ci` in the extension directory (see README). Psalm is ready to be run as well, but is disabled for now since it seems to glitch out due to too many issues. It certainly was slowing me down more than helping. Code coverage also not working yet. Apparently MW 37 comes with a 3 year old version of PHPUnit that does not support code coverage on PHP 8... wtf 

Possible next steps:

* Adjust Special:Export once requirements are clarified
* Refactor away duplication in the import code
* Refactor functions in ApiUtils and Utils into well-designed services and cover with tests
* See if we can get decent high level tests for the special pages

**Comment by rht** [comment]
Impressive progress!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -76,8 +76,15 @@ jobs:
           php maintenance/update.php --quick
 
       - name: Run PHPUnit
+        run: php tests/phpunit/phpunit.php -c extensions/DataAccounting
+        if: matrix.mw != 'REL1_37'
+
+      - name: Run PHPUnit with code coverage
         run: |
-          php tests/phpunit/phpunit.php -c extensions/DataAccounting
+          php tests/phpunit/phpunit.php -c extensions/DataAccounting/ --coverage-clover coverage.xml
```
Comment (rht) [inline .github/workflows/ci.yml:84]: I think there is a GH Actions module for coverage badge. Should we use Actions instead?

Comment (JeroenDeDauw) [reply .github/workflows/ci.yml:84]: I have not seen that approach yet. Do you think it is better in some manner than these two lines?

Comment (rht) [reply .github/workflows/ci.yml:84]: I actually have just read https://github.com/codecov/codecov-action. It is only convenient to use Actions if more info needs to be provided to codecov.io, e.g. commit hash. Probably overkill with Actions.

**Pull request merged**